src/components/FeedbackSticky/index.tsx

```diff
@@ -0,0 +1,203 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import {
+  FeedbackContainer,
+  VoteButton,
+  VoteButtonReplace,
+  VoteButtonsContainer,
+  FeedbackText,
+  Divider,
+  ButtonStyles
+} from './styles';
+import { trackFeedbackSubmission } from '../../utils/track';
+import { Icon, Button } from '@cloudscape-design/components';
+
+enum FeedbackState {
+  START = 'START',
+  UP = 'UP',
+  DOWN = 'DOWN',
+  HIDDEN = 'HIDDEN'
+}
+
+type Feedback = {
+  vote: boolean;
+  page_path: string;
+  id?: string;
+  comment?: string;
+};
+
+export default function Feedback({ footerRef }) {
+  const [state, setState] = useState<FeedbackState>(FeedbackState.START);
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const footer = footerRef.current;
+
+    if (typeof window !== 'undefined') {
+      window.addEventListener('touchmove', hideFeedback);
+      window.addEventListener('scroll', hideFeedback);
+
+      let prevScrollPos = window.scrollY;
+
+      // eslint-disable-next-line no-inner-declarations
+      function hideFeedback() {
+        // Scroll variables and calculations
+        const currPos = window.scrollY,
+          footerVisible =
+            document.body.scrollHeight - footer.offsetHeight + 50 <=
```

Comment:
What does the 50px offset included in this statement represent?  Can we use a variable with a descriptive name here instead?

Reply:
Good idea. It's so that if a bit of the footer is visible, it doesn't prevent the component from coming back in.